### ISSUE TITLE
Proper implementation of `sslmode=verify-full`

### ISSUE THREAD

**Issue by cyangle** [will/crystal-pg#240]
Given a lot of people use `cleartext` auth method to connect to DBs hosted by Azure and CockroachLabs.

I think it's time to implement `sslmode=verify-full` to protect these use cases.

**Comment by will** [comment]
Yes, this would be a good feature, and I'd be happy to help someone write it.

**Comment by steve-bitdotio** [comment]
Does crystal-pg support `sslmode=require`? I am currently unable to connect to a database that requires SSL via crystal-pg and I am wondering if the issue is related.

**Comment by plambert** [comment]
I think I might take a stab at this--I could certainly use it in one of my homelab projects, and I'm assuming it would be helpful to others.

**Comment by plambert** [comment]
> I think I might take a stab at this--I could certainly use it in one of my homelab projects, and I'm assuming it would be helpful to others.

I've opened https://github.com/will/crystal-pg/pull/311 with my changes, and would love feedback!
